src/app/random-name/random-name.component.ts

```diff
@@ -1,24 +1,35 @@
 import { Component, OnInit } from '@angular/core';
 import { RandomNameService } from './random-name.service';
+import { Observable } from 'rxjs';
+import { map, catchError } from 'rxjs/operators';
+import { Name, NamesData } from './names';
 
 @Component({
   selector: 'app-random-name',
   templateUrl: './random-name.component.html',
   styleUrls: ['./random-name.component.css']
 })
 export class RandomNameComponent implements OnInit {
-  public generatedName: string;
+  public errorMessage: string = '';
+  public $generatedName: Observable<Name>;
 
-  constructor(private randomName: RandomNameService) { }
+  constructor(private randomNameService: RandomNameService) {}
 
-  public showGeneratedName() {
-    this.randomName.getNames()
-      .subscribe((data: any[]) => {
-        this.generatedName = data[Math.floor(Math.random() * data.length)];
-      });
+  ngOnInit() {
+    this.load();
   }
 
-  ngOnInit() {
-    this.showGeneratedName();
+  load() {
+    this.$generatedName = this.randomNameService
+      .getNames()
+      .pipe(
+        map((names: NamesData) => {
```

Comment:
Tutaj zdecydowanie pokazujesz, że ten interfejs to nie jest to co zdefiniowałeś wyżej, bo korzystasz z names[i] a nie names.data, które tu nie istnieje. A pole lenght jest składową tablicy a nie NamesData.

Reply:
poprawiłem na Name[] (czyli tablicę stringów, tak?)

Reply:
Tak o ile Name będzie stringiem a nie obiektem, którym jest obecnej definicji. A dokładniej:
Ty masz: 
interface Name { value: string}, a co za tym idzie oczekujesz tablicy: names: Name[] = [{value: 'Michał}, {value: 'Wojtek']]
Powinieneś mieć:
type Name = string;
i wtedy oczekujesz tablicy names: Name[] = ['Michał','Wojtek'];

Reply:
Teraz to do mnie w końcu dotarło! Dzięki :)